### PR TITLE
[ticket/16277] Move from each() function

### PR DESCRIPTION
**Pull request by rxu**
Checklist:

- [x] Correct branch: master for new features; 3.3.x & 3.2.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/dev/master/development/coding_guidelines.html), [3.3.x](https://area51.phpbb.com/docs/dev/3.3.x/development/coding_guidelines.html) and [3.2.x](https://area51.phpbb.com/docs/dev/3.2.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.3.x/development/git.html)

Tracker ticket (set the ticket ID to **your ticket ID**):

<a href="https://tracker.phpbb.com/browse/PHPBB3-16277">PHPBB3-16277</a>.

### CODE REVIEW

**Comment by 3D-I** [comment]
`ROOT\phpbb\request\type_cast_helper.php` at line 109, is it ok?

**Comment by rxu** [comment]
@3D-I ?

**Comment by 3D-I** [comment]
I mean this line is ok or should it be fixed?

https://github.com/phpbb/phpbb/blob/036871cd9535503f1f7456096e6368544bdd8fc5/phpBB/phpbb/request/type_cast_helper.php#L109

**Comment by rxu** [comment]
Ah. It is alredy in this PR? https://github.com/phpbb/phpbb/pull/5798/files#diff-e45a34a00aaa203b1577640fbed16597R86-R124

**Comment by 3D-I** [comment]
> **Edit:** sorry @rxu I missed you already did it.

Yeah sorry, I saw it 10 minutes ago.

File: phpBB/includes/diff/engine.php [hunk 1]
```diff
@@ -285,7 +285,7 @@ function _diag($xoff, $xlim, $yoff, $ylim, $nchunks)
 				$matches = $ymatches[$line];
 
 				reset($matches);
-				while (list(, $y) = each($matches))
+				while ($y = current($matches))
```
Comment (marc1706) [inline phpBB/includes/diff/engine.php]: That's not the equivalent. `current()` will not move the pointer and therefore will always return the first element in this case.

File: phpBB/includes/diff/engine.php [hunk 2]
```diff
@@ -296,7 +296,7 @@ function _diag($xoff, $xlim, $yoff, $ylim, $nchunks)
 				}
 
 				// no reset() here
-				while (list(, $y) = each($matches))
+				while ($y = current($matches))
```
Comment (marc1706) [inline phpBB/includes/diff/engine.php]: Same as above.

File: phpBB/includes/diff/engine.php (hunk 1 above)
Comment (marc1706) [reply phpBB/includes/diff/engine.php]: I think a simple `foreach ($matches as $y)` would be the correct replacement.

**Comment by mrgoldy** [comment]
You can also have a look at my CP PR's, as I've fixed all of these in there already.

**Comment by rxu** [comment]
Thanks. I just was under impression that `current()` moves array pointer forward. I guess I misread the manual.

File: phpBB/includes/diff/engine.php (hunk 1 above)
Comment (rxu) [reply phpBB/includes/diff/engine.php]: The issue with `foreach` is that this 1st `while` loop is to be broken on `empty($this->in_seq[$y])` and the 2nd `while` loop (right below) starts with no pointer reset from the current array position.
I wonder what's gonna happen if we use `foreach` for both.

Comment (rxu) [reply phpBB/includes/diff/engine.php]: The logic should be restored now.

**Pull request merged**